front/src/modules/rollingStock/components/RollingStockSelector/RollingStockCurve.tsx

```diff
@@ -106,17 +119,36 @@ function Legend(props: {
           tabIndex={0}
           onClick={() => changeCurveState(curve.id)}
         >
-          {curve.mode}
-          {curve.comfort !== 'STANDARD' ? comfort2pictogram(curve.comfort) : null}
+          {isOnEditionMode && showPowerRestriction && curve.power_restriction}
+          {isOnEditionMode && !showPowerRestriction && curve.electrical_profile_level}
+          {!isOnEditionMode && !showPowerRestriction && curve.mode}
+          {curve.comfort !== STANDARD_COMFORT_LEVEL && !isOnEditionMode
+            ? comfort2pictogram(curve.comfort)
+            : null}
         </span>
       ))}
     </span>
   );
 }
 
 // Choose cyclic color for curves
-function curveColor(index: number) {
+function curveColor(
```

Comment:
This comment type is better because it creates documentation (you can read it when you hover the function later in the code :) )

Also, in a future ticket, it would be nice to separate RollingStockCurve in an other repository and to move this function to a file utils
```suggestion
/** Choose cyclic color for curves depending on curve number*/
function curveColor(
```

Reply:
Very useful, thx !